### DISCUSSION TITLE
Open related table links in new tab

### DISCUSSION THREAD

**Discussion by krilim** [dbeaver/dbeaver#16609] [Q&A] [answered]
When a table is related to another one, a link appears in the cell to open the related row in the other table.

Can it be opened in another tab or, at least, give than option in context menu?

Thanks.

**Comment by LonwoLonwo** [comment]
Can you provide some example please? With DDL.

**Comment by krilim** [comment]
Ie.

Two tables: **_user_** and _**location**_.

DDLs are:

```
-- praxdate.`user` definition

CREATE TABLE `user` (
  `id` bigint(20) unsigned NOT NULL AUTO_INCREMENT,
  `uuid` char(36) NOT NULL,
  `uuid_company_default` char(36) NOT NULL,
  `uuid_location_default` char(36) DEFAULT NULL,
  `name` varchar(100) NOT NULL,
  `lastname` varchar(100) NOT NULL,
  `email` varchar(100) NOT NULL,
  `password` char(64) NOT NULL,
  `rooms_view` tinyint(1) unsigned NOT NULL DEFAULT '0',
  `superuser` tinyint(1) unsigned NOT NULL DEFAULT '0',
  `active` tinyint(1) unsigned NOT NULL DEFAULT '1',
  `session_id` varchar(50) DEFAULT NULL,
  `login_link_token` char(96) DEFAULT NULL,
  `last_broadcast` bigint(20) unsigned DEFAULT NULL,
  `last_login` timestamp NULL DEFAULT NULL,
  `created_at` timestamp NULL DEFAULT NULL,
  `updated_at` timestamp NULL DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `user_UN` (`uuid`),
  UNIQUE KEY `user_UN_email` (`email`),
  KEY `user_uuid_company_default_IDX` (`uuid_company_default`) USING BTREE,
  KEY `user_FK_broadcast` (`last_broadcast`),
  KEY `user_FK_location` (`uuid_location_default`),
  CONSTRAINT `user_FK_broadcast` FOREIGN KEY (`last_broadcast`) REFERENCES `broadcast` (`id`) ON UPDATE CASCADE,
  CONSTRAINT `user_FK_company` FOREIGN KEY (`uuid_company_default`) REFERENCES `company` (`uuid`) ON DELETE NO ACTION ON UPDATE CASCADE,
  CONSTRAINT `user_FK_location` FOREIGN KEY (`uuid_location_default`) REFERENCES `location` (`uuid`) ON UPDATE CASCADE
) ENGINE=InnoDB AUTO_INCREMENT=10 DEFAULT CHARSET=utf8mb4;
```

```
-- praxdate.location definition

CREATE TABLE `location` (
  `uuid` char(36) NOT NULL,
  `uuid_company` char(36) NOT NULL,
  `active` tinyint(1) NOT NULL DEFAULT '1',
  `name` varchar(100) NOT NULL,
  `url` varchar(250) DEFAULT NULL,
  `email` varchar(100) DEFAULT NULL,
  `address` tinytext,
  `city` varchar(100) DEFAULT NULL,
  `province` varchar(100) DEFAULT NULL,
  `zip_code` varchar(5) DEFAULT NULL,
  `telephone` varchar(25) DEFAULT NULL,
  `vat_number` varchar(100) DEFAULT NULL,
  `sms_autoreply_message` text,
  `sms_template` text,
  `whatsapp_autoreply_message` text,
  `whatsapp_template` text,
  `next_invoice_number` bigint(20) unsigned DEFAULT NULL,
  `current_invoice_prefix` varchar(50) DEFAULT NULL,
  `stats_configuration` longblob,
  `created_at` timestamp NULL DEFAULT NULL,
  `updated_at` timestamp NULL DEFAULT NULL,
  PRIMARY KEY (`uuid`),
  UNIQUE KEY `location_UN` (`uuid_company`,`name`),
  CONSTRAINT `location_FK_company` FOREIGN KEY (`uuid_company`) REFERENCES `company` (`uuid`) ON DELETE CASCADE ON UPDATE CASCADE
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4;
```

When I browse data in **_user_** table, column _uuid_location_default_ is linked to row with that _uuid_ in table _**location**_.

**Comment by Xylandor** [comment, marked as answer]
@krilim

If I'm understanding correctly, when you click on the referenced link in either the table data view or in the query results list, instead of it replacing the current tab, open it in a new tab. This functionality already exists and can be done by holding down COMMAND (Mac) or CONTROL (Win) and clicking the link. This opens it in a new tab (see attached video).

It would be nice, however, to have a preference option that allowed us to control this behavior; whether to open in the same tab or new tab.

https://user-images.githubusercontent.com/24617679/170362326-7c91ac33-0458-4095-8cc3-2e35940318de.mov

**Comment by krilim** [comment]
Thanks so much :-D

**Comment by ThisIsABug** [comment]
Sorry for bump, but did this feature request ( https://github.com/dbeaver/dbeaver/issues/16548 ) ever get implemented in any form?

**Comment by krilim** [comment]
I saw the video about using CMD on Mac ... but that does not work for mine. Even if I push CMD together the new data replace the original one (see video) . using version 24.0.5.202405191503

https://github.com/dbeaver/dbeaver/assets/2741646/dd63a47c-7517-4387-a719-83a4ef313224

**Reply by 4javier** [reply]
I think that column is a self referencing property. It links to an entry into the same table.
If that's the case, it sounds reasonable to me that it behaves that way: there's already a tab opened for that  table, so it just filter existing data. 
It does the same if you try to ctrl+click on a link for a different table already opened in another tab: it will not create a second tab for the table, it will just move the already opened one and filter the entries.
